cleanup/cleanDockerImages/cleanDockerImages.groovy

```diff
@@ -0,0 +1,135 @@
+        repoList = repoList.sort { it[1] }
+        def deleteCount = repoList.size() - maxImagesCount
+        for (i = 0; i < deleteCount; i += 1) {
+            deleted << repoList[i][0].id
+            if (!dryRun) repositories.delete(repoList[i][0])
+        }
+    }
+    return deleted
+}
+
+// Traverse through the docker repo (directories and sub-directories) and:
+// - delete the images immediately if the maxDays policy applies
+// - Aggregate the images that qualify for maxCount policy (to get deleted in
+//   the execution closure)
+def simpleTraverse(parentInfo, oldSet, imagesPathMap, imagesCount, daysToKeep) {
+    def maxCount = null
+    def parentRepoPath = parentInfo.repoPath
+    for (childItem in repositories.getChildren(parentRepoPath)) {
+        def currentPath = childItem.repoPath
+        if (childItem.isFolder()) {
+            simpleTraverse(childItem, oldSet, imagesPathMap, imagesCount, daysToKeep)
+            continue
+        }
+        log.debug("Scanning File: $currentPath.name")
+        if (currentPath.name != "manifest.json") continue
+        // get the properties here and delete based on policies:
+        // - implement daysPassed policy first and delete the images that
+        //   qualify
+        // - aggregate the image info to group by image and sort by create
+        //   date for maxCount policy
+        if (checkDaysPassedForDelete(childItem, daysToKeep)) {
+            log.debug("Adding to OLD MAP: $parentRepoPath")
+            oldSet << parentRepoPath
+        } else if ((maxCount = getMaxCountForDelete(childItem)) > 0) {
+            log.debug("Adding to IMAGES MAP: $parentRepoPath")
+            def parentCreatedDate = parentInfo.created
+            def parentId = parentRepoPath.parent.id
+            def oldmax = maxCount
+            if (parentId in imagesCount) oldmax = imagesCount[parentId]
+            imagesCount[parentId] = maxCount > oldmax ? maxCount : oldmax
+            if (!imagesPathMap.containsKey(parentId)) {
+                imagesPathMap[parentId] = []
+            }
+            imagesPathMap[parentId] << [parentRepoPath, childItem.created]
+        }
+        break
+    }
+}
+
+// This method checks if the docker image's manifest has the property
+// "docker.label.com.jfrog.artifactory.retention.maxDays" for purge.
+// If not found, number_of_days_to_keep from properties file will be used instead.
+
+def checkDaysPassedForDelete(item, daysToKeep) {
+    def maxDaysProp = "docker.label.com.jfrog.artifactory.retention.maxDays"
+    def oneday = TimeUnit.MILLISECONDS.convert(1, TimeUnit.DAYS)
+    def prop = repositories.getProperty(item.repoPath, maxDaysProp)
+    if (!prop && (daysToKeep == null)) return false
+        else if (((new Date().time - item.created) / oneday) >= daysToKeep) return true  
+    log.debug("PROPERTY $maxDaysProp FOUND = $prop IN MANIFEST FILE")
+    prop = prop.isInteger() ? prop.toInteger() : null
```

Comment:
Isn't `toInteger()` returning null if it cannot be converted to an int?

Reply:
I don't know :) This is not mine :)